### PR TITLE
[master] Avoid suggesting getter setter code actions when functions already exists

### LINKED ISSUE #36652: Getter Setter CodeActions provides code actions when getter and setter are already defined
**Description:**

Getter Setter CodeActions provides code actions when getter and setter already defined

**Steps to reproduce:**

consider following ocde example

```ballerina
class Person {
    private int age = 5;

    public function getAge() returns int { 
        return self.age;
    }

    public function setAge(int  age) { 
        self.age = age;
    }    
}

```
Follow below steps

1. edit field `private int age = 5` to `private int age`
2. perform getter setter codeaction again

### PR DESCRIPTION
**Pull request by DamithaSenevirathne**
## Purpose

Fixes #36652

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/test/resources/codeaction/getter-setter/source/getterSetter5.bal [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Person {
+
+    private int age   ;
```
Comment (TharushiJay) [inline language-server/modules/langserver-core/src/test/resources/codeaction/getter-setter/source/getterSetter5.bal:3]:
```suggestion
    private int age;
```

File: language-server/modules/langserver-core/src/test/resources/codeaction/getter-setter/source/getterSetter5.bal [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+    public function getAge() returns int {
+        return self.age;
+    }
+
+    public function setAge(int age) {
+        self.age = age;
+    }
+
+}
```
Comment (TharushiJay) [inline language-server/modules/langserver-core/src/test/resources/codeaction/getter-setter/source/getterSetter5.bal:13]:
```suggestion
}

```

File: language-server/modules/langserver-core/src/test/resources/codeaction/getter-setter/config/getterSetterCodeAction5.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "position": {
+    "line": 2,
+    "character": 17
+  },
+  "source": "getterSetter5.bal",
+
+  "expected": [
+
+  ]
```
Comment (TharushiJay) [inline language-server/modules/langserver-core/src/test/resources/codeaction/getter-setter/config/getterSetterCodeAction5.json:10]:
```suggestion
  "expected": []
```

**Review by TharushiJay — COMMENTED** [review]
Changes look good. Added some minor formatting suggestions.

**Review by gayalkuruppu — CHANGES_REQUESTED** [review]
Shall improve following cases?

1. Add a space between type and the opening brace in getter function
```
    public function getAge() returns int{ 
        return self.age;
    }
```

2. improve the code action to work when the field name is a quotedIdentifier

**Comment by mohanvive** [comment]
@DamithaSenevirathne can you please accommodate these review comments?

File: language-server/modules/langserver-core/src/test/resources/codeaction/getter-setter/source/getterSetter5.bal (hunk 2 above)
Comment (DamithaSenevirathne) [reply language-server/modules/langserver-core/src/test/resources/codeaction/getter-setter/source/getterSetter5.bal]: Updated in 346328f74b6b1c25f5a6e0a93966e27bd829d265

File: language-server/modules/langserver-core/src/test/resources/codeaction/getter-setter/config/getterSetterCodeAction5.json (hunk 3 above)
Comment (DamithaSenevirathne) [reply language-server/modules/langserver-core/src/test/resources/codeaction/getter-setter/config/getterSetterCodeAction5.json]: Updated in 74a8579fa2abf778fd506a18a96a4ff6239865fc

**Comment by DamithaSenevirathne** [comment]
> Shall we improve following cases?
> 
>     1. Add a space between type and the opening brace in getter function
> 
> 
> ```
>     public function getAge() returns int{ 
>         return self.age;
>     }
> ```
> 
>     2. improve the code action to work when the field name is a quotedIdentifier

Updated in f6fb2ae65307c9b50a7326a70f1f878d66eec0b3

File: language-server/modules/langserver-core/src/test/resources/codeaction/getter-setter/source/getterSetter6.bal [hunk 4]
```diff
@@ -0,0 +1,6 @@
+public class Counter {
+
+    private int 'count;
+    private string empName;
+
```
Comment (LakshanWeerasinghe) [inline language-server/modules/langserver-core/src/test/resources/codeaction/getter-setter/source/getterSetter6.bal:5]: If we add the below code segment and what would be the suggested code actions for `'count`?
It should suggest a different getter signature.
```
public function getCount() returns int {
    return 5;
}
```

Comment (gayalkuruppu) [reply language-server/modules/langserver-core/src/test/resources/codeaction/getter-setter/source/getterSetter6.bal:5]: Here in this case, we already have a getter for `'count` hence we will only suggest the setter.

**Comment by DamithaSenevirathne** [comment]
Merged to 2201.2.x. and synced to the master https://github.com/ballerina-platform/ballerina-lang/pull/37813

**Pull request closed without merging**